project1/server/server_tasks/agent_task.h

```diff
@@ -0,0 +1,56 @@
+/**
+ * @File Task wrapper for the Agent Class.
+ */
+
+#ifndef PROJECT1_AGENT_TASK_H
+#define PROJECT1_AGENT_TASK_H
+
+#include "../../thread_pool/task.h"
+#include "../../thread_pool/thread_pool.h"
+#include "../file_handler/file_handler.h"
+#include "../agent.h"
+#include "command_ids.h"
+
+namespace server_tasks {
+    class AgentTask : public thread_pool::Task {
+    public:
+        thread_pool::Task::Status RunAtomic() final;
+        thread_pool::Task::Status SetUp();
+        void CleanUp() final;
+
+        /**
+         * @brief Sets the ClientFD for this agent.
+         * @param id
+         * @note Needs to be set before being added to the thread pool.
+         */
+        void SetClientFD(int id);
```

Comment:
Again, can this not just be passed in the constructor?

Reply:
I'll change these